src/celengine/galaxy.cpp

```diff
@@ -128,192 +371,115 @@ void Galaxy::setDetail(float d)
 }
 
 
-void Galaxy::setCustomTmpName(const string& tmpNameStr)
+void Galaxy::setCustomTmpName(std::string&& tmpNameStr)
```

Comment:
maybe just std::string? (or even std::string_view?).

Reply:
This is essentially a sink method: I prefer to consider things like `std::string` and `std::vector` as "morally" like `std::unique_ptr`, and have the API set up to do moves rather than copy. Most of the time the calling code doesn't need the value afterwards anyway, so copying (which would happen with `const std::string&`) incurs unnecessary allocations.

Nevertheless, this method is completely unnecessary and going away.